### PR TITLE
[SL-UP] Adds and updates logs for RS911X

### PR DESCRIPTION
**Pull request by rosahay-silabs**
This PR aims to add error status codes for RS911x with respect to MATTER-4193

### CODE REVIEW

**Review by mkardous-silabs — APPROVED** [review]
If are updating the error / progress level of certain logs, can we double check the other prints to make sure they have the correct level as well?

File: src/platform/silabs/wifi/rs911x/WifiInterface.cpp
Comment (mkardous-silabs) [inline src/platform/silabs/wifi/rs911x/WifiInterface.cpp:1]: What is the change in this file? This doesn't seem related to what the initial goal of the PR was?

Comment (mkardous-silabs) [reply src/platform/silabs/wifi/rs911x/WifiInterface.cpp:1]: ok thanks for the information! I was surprise to see the VerifyOrReturn being added - seemed like a bug fix

**Review by mkardous-silabs — APPROVED** [review]

**Review by chirag-silabs — APPROVED** [review]

**Review by senthilku — APPROVED** [review]
